wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -301,6 +337,9 @@ static String sanitizeJavadoc(String documentation) {
   /** Returns the generated code for {@code type}, which may be a top-level or a nested type. */
   public TypeSpec generateType(Type type) {
     if (type instanceof MessageType) {
+      if (protoTypeToAdapter.containsKey(type.type())) {
+        return generateProtoFields((MessageType) type);
```

Comment:
it’s lame that you made the map lookup on line 340, and then the very first thing the implementation of this method must do is also do the map lookup. Maybe instead this method should take the ClassName that it’s going to generate?